### PR TITLE
Mob 3534

### PR DESCRIPTION
**Pull request by dianakarh**
## JIRA
[MOB-3534](https://karhoo.atlassian.net/browse/MOB-3534)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [ ] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [x] Other: add support for standalone use of the passenger details page

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: KarhooUISDK/Screens/PassengerDetails/PassengerDetailsViewController.swift [hunk 1]
```diff
@@ -313,13 +329,13 @@ extension PassengerDetailsViewController: KarhooInputViewDelegate {
     private func getValidInputViewCount() -> Int {
         var validSet = Set<String>()
         
-        for (_, inputView) in inputViews.enumerated() {
+        inputViews.forEach({ inputView in
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDK/Screens/PassengerDetails/PassengerDetailsViewController.swift:332]: `()` redundant

File: KarhooUISDK/Screens/PassengerDetails/PassengerDetailsViewController.swift [hunk 2]
```diff
@@ -313,13 +329,13 @@ extension PassengerDetailsViewController: KarhooInputViewDelegate {
...
             if inputView.isValid() {
                 validSet.insert(inputView.accessibilityIdentifier!)
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDK/Screens/PassengerDetails/PassengerDetailsViewController.swift:334]: Force unwrapping in not necessary here. I'd add default value ("") just to limit potential production crash cases.

File: KarhooUISDK/Screens/CheckoutScreen/Checkout/KarhooCheckoutPresenter.swift [hunk 3]
```diff
@@ -505,3 +496,14 @@ final class KarhooCheckoutPresenter: CheckoutPresenter {
...
 }
+
+extension KarhooCheckoutPresenter: PassengerDetailsDelegate {
+    func didInputPassengerDetails(result: PassengerDetailsResult) {
+        view?.setPassenger(details: result.details)
+        PassengerInfo.shared.set(country: result.country ?? KarhooCountryParser.defaultCountry)
+    }
+    
+    func didCancelInput(byUser: Bool) {
+
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDK/Screens/CheckoutScreen/Checkout/KarhooCheckoutPresenter.swift:507]: Maybe add some comment to inform this method is empty on purpose.

**Review by AleksanderWedrychowskiKarhoo — CHANGES_REQUESTED** [review]
Small cleanup required but related only to formatting.

**Review by mohadian — APPROVED** [review]

File: KarhooUISDK/Screens/PassengerDetails/PassengerDetailsViewController.swift [hunk 4]
```diff
@@ -39,6 +39,25 @@ final class PassengerDetailsViewController: UIViewController, BaseViewController
     private var shouldMoveToNextInputViewOnReturn = true
     private let currentLocale = NSLocale.current.languageCode ?? "en"
     
+    // MARK: - PassengerDetailsView
+    var delegate: PassengerDetailsDelegate? {
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDK/Screens/PassengerDetails/PassengerDetailsViewController.swift:43]: Strong reference cycle

File: KarhooUISDK/Screens/PassengerDetails/PassengerDetailsViewController.swift [hunk 5]
```diff
@@ -266,7 +282,7 @@ final class PassengerDetailsViewController: UIViewController, BaseViewController
     }
     
     func addPreInputtedDetails() {
-        guard let details = presenter.details
+        guard let details = details
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDK/Screens/PassengerDetails/PassengerDetailsViewController.swift:285]: Imho it should be one-line guard statement to keep it simple

**Review by AleksanderWedrychowskiKarhoo — CHANGES_REQUESTED** [review]

File: KarhooUISDK/Screens/PassengerDetails/PassengerDetailsViewController.swift (hunk 4 above)
Comment (dianakarh) [reply KarhooUISDK/Screens/PassengerDetails/PassengerDetailsViewController.swift:43]: Fixed in the `PassengerDetailsPresenter`

**Review by AleksanderWedrychowskiKarhoo — APPROVED** [review]

**Pull request merged**